### PR TITLE
Release on github with gh-action

### PR DESCRIPTION
**Pull request by wzshiming**
I'm going to put a tag `v0.0.1-alpha.0` on this branch and see if the GH action can be triggered

/hold
/cc @Huang-Wei @yibozhuang

### CODE REVIEW

File: .github/workflows/releases.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: Releases
+on:
+  push:
+    tags:
+      - v*
+
+env:
+  PUSH: 'true'
+  DIRECT: 'true'
+  GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Huang-Wei) [inline .github/workflows/releases.yaml:10]: seems it's not used?

File: .github/workflows/releases.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.18
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Log into registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
```
Comment (Huang-Wei) [inline .github/workflows/releases.yaml:26]: qq: does GH provide a native directive for login? Not sure this plain `run` directive may leak the token? like display in the log publicly accessible?

**Review by Huang-Wei — COMMENTED** [review]
some nits.

File: .github/workflows/releases.yaml (hunk 1 above)
Comment (wzshiming) [reply .github/workflows/releases.yaml:10]: `gh release upload` will use `GH_TOKEN` to put `binary` into the release page.

File: .github/workflows/releases.yaml (hunk 2 above)
Comment (wzshiming) [reply .github/workflows/releases.yaml:26]: Updated

File: .github/workflows/releases.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  releases:
+    runs-on: ubuntu-latest
+    if: github.event_name != 'pull_request'
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.18
+      - name: Set up QEMU
```
Comment (yibozhuang) [inline .github/workflows/releases.yaml:22]: QEMU is for cross-compile correct?

**Review by yibozhuang — APPROVED** [review]

File: .github/workflows/releases.yaml (hunk 3 above)
Comment (wzshiming) [reply .github/workflows/releases.yaml:22]: Is for building multi-architecture images.

Comment (wzshiming) [reply .github/workflows/releases.yaml:22]: https://github.com/docker/setup-qemu-action#about

**Pull request merged**